docs/rfcs/1013-parsec-account.md

```diff
@@ -0,0 +1,507 @@
+
+  We assume that the user should be able to connect from any browser/computer.
+
+That is how we came to the idea of storing the device on a remote server.
+
+- That will solve the issue of the browser cleaning its data.
+- The devices would be accessible from anywhere (given the user is online).
+
+## Goals
+
+- Describe how to store/access the user devices on a remote service.
+- Propose an API to implement this feature.
+- Focus on allowing users to connect using a password
+
+## Non-Goals
+
+- Discuss security concerns regarding the web client application.
+- How the devices are created (nothing new, the creation is performed as usual by the client)
+- Support authentication methods other than password (will be addressed in a new RFC)
+
+## Design
+
+### General principles
+
+An authentication service (Parsec Account) would be used to store user devices.
+This service will be requested before connecting to the metadata server (Parsec server) in order to retrieve a specific user device.
+The device will then be used to authenticate to the metadata server.
+
+<!-- Device stored on a third-party service -->
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Client
+  participant S_Acc as Account server
+  participant S_Data as Metadata server
+
+  Client ->> S_Acc: Retrieve the device of the user
+  S_Acc ->> Client: Device of the user
+
+  Client ->> S_Data: Exchange with the server
+```
+
+### Glossary
+
+- **Parsec Account:** The name of the new service that will store the devices for a user.
+- **Account:** The user's account inside Parsec account.
+- **Vault:**: A pool of encrypted data accessible to a group of authentication methods.
+- **Protected device**: The devices that the user want to protect.
+
+- `auth_method_master_key`: A secret obtained from the authentication method.
+- `auth_method_hmac_key`: A secret used for HMAC based authentication.
+- `auth_method_secret_key`: A symmetric key to encrypt the vault key
+- `vault_key`: A symmetric key used to encrypt the vault.
+
+### Key generation
+
+- `auth_method_master_key` is generated from a password using a PBKDF algorithm (likely `argon2`).
+- `auth_method_hmac_key` and `auth_method_secret_key` are generated from `auth_method_master_key` using a KDF algorithm (the algorithm will be configured differently to not use the same key).
+- `vault_key` is generated from a CSPRNG.
+
+
+### How we authenticate the request
+
+Some indices are already present above, we will use an HMAC based authentication.
+
+> The reason for this choice is that we do not need additional request to get an OAuth token.
+
+The HMAC code will be put in the `Authorization` HTTP header and will have the following format:
+
+```bash
+PARSEC-HMAC-BLAKE2B.${id}.${account_email}.${timestamp}.${signature}
+```
+
+The signature is generated like so:
+
+```math
+\begin{gather}
+body\_sha256 = sha256(body) \\
+content = \text{"PARSEC-HMAC-BLAKE2B"} | email | timestamp | body\_sha256 \\
```

Comment:
What does | mean in this context ?

Reply:
It's the concatenation operator but wikipedia says it should be `||` instead.
So I will fix this

Reply:
done